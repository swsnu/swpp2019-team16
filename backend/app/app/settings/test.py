from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
