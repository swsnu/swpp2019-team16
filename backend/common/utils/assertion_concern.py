class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
