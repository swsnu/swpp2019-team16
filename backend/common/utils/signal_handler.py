import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
