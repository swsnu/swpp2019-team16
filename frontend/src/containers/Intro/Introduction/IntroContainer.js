import React, { useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import Introduction from '../../../components/Intro/Introduction';

function IntroContainer({ history }) {
  const onStart = useCallback(() => {
    history.push('/request');
  }, [history]);
  return <Introduction onStart={onStart} />;
}
export default withRouter(IntroContainer);
