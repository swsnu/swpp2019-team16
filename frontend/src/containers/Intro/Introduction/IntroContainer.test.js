import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import IntroContainer from './IntroContainer';
import { renderWithRedux } from 'test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<IntroContainer />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/intro']}>
        <Route component={IntroContainer} path="/intro" />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
