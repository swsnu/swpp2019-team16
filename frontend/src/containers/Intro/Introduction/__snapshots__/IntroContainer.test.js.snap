// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroContainer /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="Introduction"
  >
    <p
      class="MuiTypography-root MuiTypography-h1 MuiTypography-paragraph"
    >
      Welcome to YA-TA!
    </p>
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      What is yata?
    </h3>
    <p
      class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph"
    >
      Ya-Ta helps students who need to urgently get to class. Within ten minutes, Ya-Ta groups students with these similar, urgent, needs, and grabs a taxi for them.
    </p>
    <div
      class="startButton"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Start!
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
